### PR TITLE
MULE-18921: Migrate xml to declaration logic to use the AST instead of ConfigLine

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/XmlArtifactDeclarationLoader.java [hunk 1]
```diff
@@ -20,6 +24,32 @@
...
 public interface XmlArtifactDeclarationLoader {
 
+  public static final class Initializer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Initializer.class);
+
+    private static Class<?> LOADER_CLASS;
+
+    static {
+      String loaderClassName = System.getProperty(XmlArtifactDeclarationLoader.class.getName() + ".loaderClassName");
```
Comment (mlischetti) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/XmlArtifactDeclarationLoader.java:34]: Use static import

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/XmlArtifactDeclarationLoader.java [hunk 2]
```diff
@@ -20,6 +24,32 @@
  */
 public interface XmlArtifactDeclarationLoader {
 
+  public static final class Initializer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Initializer.class);
```
Comment (mlischetti) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/XmlArtifactDeclarationLoader.java:29]: static import

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/InfrastructureElementModelDelegate.java [hunk 3]
```diff
@@ -334,6 +326,7 @@ private void createStreamingStrategy(ParameterValue value,
                               namespace);
   }
 
+  // TODO review if this is needed
```
Comment (mlischetti) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/InfrastructureElementModelDelegate.java:329]: a mule ticket is need for that?

Comment (elrodro83) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/model/InfrastructureElementModelDelegate.java:329]: nope, just something i forgot to remove

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/SchemaConstants.java [hunk 4]
```diff
@@ -95,4 +104,16 @@
   public static final String CURRENT_VERSION = "current";
 
   private SchemaConstants() {}
+
+  public static Optional<String> buildRawParamKeyForDocAttribute(Entry<String, String> docAttr) {
+    final QName qName = QName.valueOf(docAttr.getKey());
+
+    if (NS_MULE_DOCUMENTATION.equals(qName.getNamespaceURI())) {
+      return of(DOC_NAMESPACE + ":" + qName.getLocalPart());
+    } else if (StringUtils.isEmpty(qName.getNamespaceURI())) {
```
Comment (mlischetti) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/SchemaConstants.java:113]: static import

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/SchemaConstants.java [hunk 5]
```diff
@@ -95,4 +104,16 @@
   public static final String CURRENT_VERSION = "current";
 
   private SchemaConstants() {}
+
+  public static Optional<String> buildRawParamKeyForDocAttribute(Entry<String, String> docAttr) {
+    final QName qName = QName.valueOf(docAttr.getKey());
```
Comment (mlischetti) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/dsl/SchemaConstants.java:109]: static import

**Comment by elrodro83** [comment]
--validate

**Pull request merged**